terraso_backend/apps/story_map/permission_rules.py

```diff
@@ -23,12 +27,56 @@ def allowed_to_view_story_map(user, story_map):
 
 @rules.predicate
 def allowed_to_change_story_map(user, story_map):
-    return story_map.created_by == user
+    is_owner = story_map.created_by == user
+    if is_owner:
+        return True
+    account_membership = (
+        story_map.membership_list.memberships.approved_only().filter(user=user).first()
+    )
+    return account_membership is not None and account_membership.user_role == ROLE_COLLABORATOR
 
 
 @rules.predicate
 def allowed_to_delete_story_map(user, story_map):
     return story_map.created_by == user
 
 
+@rules.predicate
+def allowed_to_save_story_map_membership(user, obj):
```

Comment:
question: should this be `edit` or `update`?

Reply:
@paulschreiber You mean `add` or `update` I expect this to handle both that is why I called it `save` I think I could split it if you think that is better

Reply:
What about `change`? We also have `allowed_to_change_group`, `allowed_to_change_landscape`, `allowed_to_delete_membership`?

Reply:
That makes sense to me